nimble/host/src/ble_hs_pvcy.c

```diff
@@ -63,17 +63,23 @@ int
 ble_hs_pvcy_remove_entry(uint8_t addr_type, const uint8_t *addr)
 {
     struct ble_hci_le_rmv_resolve_list_cp cmd;
+    int rc;
 
     if (addr_type > BLE_ADDR_RANDOM) {
-        return BLE_ERR_INV_HCI_CMD_PARMS;
+        addr_type = addr_type % 2;
     }
 
     cmd.peer_addr_type = addr_type;
     memcpy(cmd.peer_id_addr, addr, BLE_DEV_ADDR_LEN);
 
-    return ble_hs_hci_cmd_tx(BLE_HCI_OP(BLE_HCI_OGF_LE,
-                                        BLE_HCI_OCF_LE_RMV_RESOLV_LIST),
-                             &cmd, sizeof(cmd), NULL, 0);
+    rc = ble_hs_hci_cmd_tx(BLE_HCI_OP(BLE_HCI_OGF_LE,
```

Comment:
@prasad-alatkar  does it really can happen?

Reply:
@rymanluk actually in our case (NimBLE host only) it does happen. As per the spec,
  
>   When a Controller cannot remove a device from the resolving list because it is not found, it shall return the error code Unknown Connection Identifier (0x02).

Please let me know if I am missing anything.

Reply:
You are right, so this is valid error massage and we should just allow to send it back to the app.

Reply:
Makes sense... !! Then I might need to handle `BLE_ERR_UNK_CONN_ID` in `ble_gap_unpair` here: [return_from_ble_hs_pvcy_remove_entry](https://github.com/apache/mynewt-nimble/pull/636/files#diff-7075c75d7ac538394ce8b3302c237cdaR5201),  what do you think ?

Reply:
I would just allow it to go back as a return on ble_gap_unpair()

Reply:
But it will again block `ble_gap_unpair` to call ` ble_store_util_delete_peer`. I mean, what if the device was not present in the resolvable list and still we are trying to remove it ? From user's perspective `ble_gap_unpair` API should complete calling all the 3 APIs till `ble_store_util_delete_peer`.

Reply:
@prasad-alatkar Ah ok, I missed `ble_store_util_delete_peer` which is called after `ble_hs_pvcy_remove_entry` and of course course it will fail for devices which does not use privacy. 

In such a case I would ignore `rc` from `ble_hs_pvcy_remove_entry` at all.

Reply:
Thank you @rymanluk , will update the PR accordingly !!